### DISCUSSION TITLE
ESP32 bricked after template apply?

### DISCUSSION THREAD

**Discussion by spamik** [arendst/Tasmota#22719] [Support request] [answered]
Hi, I have custom board (made by myself) with ESP-WROOM-32 module. I've had flashed tasmota32-display 14.4.1. Everything works great. I've played with configuration -> module settings to get working things which where on the board. When I was done (and everything works), I typed template 255 to the console which gives mi this result:

`{"NAME":"Template","GPIO":[1312,1,231,1,232,419,1,1,1,1,1,1,233,234,235,416,421,420,418,417,0,224,225,236,0,228,229,230,1,1,1,1,226,227,0,1],"FLAG":0,"BASE":1}`

I've only changed name and retype it to Configuration -> Other. That was the last moment when I have seen this board working :-) When I connect uart console to board I see this errors (it's in loop):

```
00:00:00.002 HDW: ESP32-D0WDQ6-V3 v3.0 
00:00:00.006 UFS: FlashFS mounted with 308 kB free
00:00:00.013 CFG: Loaded from File, Počítej 1
00:00:00.014 FRC: Some settings have been reset (85)
00:00:00.020 Projekt tasmota - Tasmota Verze 14.4.1(TasmoCompiler-esp32generic)-3_1_0(2024-12-26T19:18:)

00:00:00.002 HDW: ESP32-D0WDQ6-V3 v3.0 
00:00:00.006 UFS: FlashFS mounted with 308 kB free
00:00:00.013 CFG: Loaded from File, Počítej 1
00:00:00.014 FRC: Some settings have been reset (86)
00:00:00.020 Projekt tasmota - Tasmota Verze 14.4.1(TasmoCompiler-esp32generic)-3_1_0(2024-12-26T19:18:)

00:00:00.002 HDW: ESP32-D0WDQ6-V3 v3.0 
00:00:00.006 UFS: FlashFS mounted with 308 kB free
00:00:00.012 CFG: Loaded from File, Počítej 1
00:00:00.014 FRC: Some settings have been reset (87)
00:00:00.020 Projekt tasmota - Tasmota Verze 14.4.1(TasmoCompiler-esp32generic)-3_1_0(2024-12-26T19:18:)
```

After that in boot mode I run esptool erase flash and writed new firmware (compiled from tasmocompiler). Write was successful but ESP still won't boot and error remains. Only the last line changed (the terminal report above was copied after this flash, previously it reports tasmota32-display version.

So do have anybody some idea what happens? If there some chance to get this chip working if the error remains even after erase_flash command?

**Comment by sfromis** [comment]
Seems to be in a boot loop it cannot recover from, as indicated by the bad message:
`FRC: Some settings have been reset (87)`
What you posted has no clues about reasons, but the last part here looks odd:
`14.4.1(TasmoCompiler-esp32generic)-3_1_0(2024-12-26T19:18:)`
Timestamp ending with a `:` and seconds missing.

Maybe try with a stepwise template buildup, to see if there is a point where it starts to happen, and notice whether that time stamp is bad before that.

**Comment by spamik** [comment]
Ok, I have found cause of this issue and it's really stupid :-) It's power issue - power supply in UART convertor is not powerful enough to full boot process. On normal supply it's working.

But still the template above break it and prevent device from booting. Now I have to figure out why (or I have to find way how to have serial console and working power supply at one time :-) )

**Comment by spamik** [comment, marked as answer]
Ok, another thing. It seems that command "Template 255" has some bug. It produced output in first post and it's apparently wrong. When I tried it again without apply check box, it's possible to see that it has wrong order. It assigned some functions to GPIO6-11 which is probably the reason why it doesn't boot anymore. After creating new module configuration manually and exporting it with "template" command it works OK.

**Reply by sfromis** [reply]
If getting from module config to a template does not work properly, that sounds like reason to create an issue detailing that.

I'd expect most people to be creating templates using Configure Template, not really using Configure Module as an extra step. I'd not even recommend that path.

**Reply by spamik** [reply]
Yup, I'll try to do it again and fill issue :-)

Otherwise you are right. I've missed configure template on first look - it was just a prototype and playing with configure module was fine (as base module was devkit). When it seems working I've started looking how to create template for it and I've tried to avoid doing configuration again... so I used merge option. But apparently it won't saved time :-)

**Comment by spamik** [comment]
Just for note, today I've played with some unused devel module to replicate problem to fill issue. But without success, now it works, I can't break it. I wonder how I managed to break it with this prototype :-) Maybe there was some race condition involved with some older firmwares (prototype was several time upgraded)...
